triton-vm/src/table/instruction_table.rs

```diff
@@ -207,21 +207,22 @@ impl ExtInstructionTable {
     }
 
     pub fn ext_transition_constraints_as_circuits(
-    ) -> Vec<ConstraintCircuit<InstructionTableChallenges>> {
-        let circuit_builder = ConstraintCircuitBuilder::new(2 * FULL_WIDTH);
-        let one: ConstraintCircuitMonad<InstructionTableChallenges> =
+    ) -> Vec<ConstraintCircuit<InstructionTableChallenges, DualRowIndicator<FULL_WIDTH>>> {
+        let circuit_builder: ConstraintCircuitBuilder<
+            InstructionTableChallenges,
+            DualRowIndicator<FULL_WIDTH>,
+        > = ConstraintCircuitBuilder::new(2 * FULL_WIDTH);
+        let one: ConstraintCircuitMonad<InstructionTableChallenges, DualRowIndicator<FULL_WIDTH>> =
             circuit_builder.b_constant(1u32.into());
-        let addr = circuit_builder.input(usize::from(Address));
-
-        let addr_next = circuit_builder.input(FULL_WIDTH + usize::from(Address));
-        let current_instruction = circuit_builder.input(usize::from(CI));
-        let current_instruction_next = circuit_builder.input(FULL_WIDTH + usize::from(CI));
-        let next_instruction = circuit_builder.input(usize::from(NIA));
-        let next_instruction_next = circuit_builder.input(FULL_WIDTH + usize::from(NIA));
-        // let is_padding_row =
-        //     one.clone() - circuit_builder.deterministic_input(FULL_WIDTH + usize::from(IsPadding));
-        let is_padding = circuit_builder.input(usize::from(IsPadding));
-        let is_padding_next = circuit_builder.input(FULL_WIDTH + usize::from(IsPadding));
+        let addr = circuit_builder.input(CurrentRow(Address.into()));
+
+        let addr_next = circuit_builder.input(NextRow(Address.into()));
+        let current_instruction = circuit_builder.input(CurrentRow(CI.into()));
+        let current_instruction_next = circuit_builder.input(NextRow(CI.into()));
+        let next_instruction = circuit_builder.input(CurrentRow(NIA.into()));
+        let next_instruction_next = circuit_builder.input(NextRow(NIA.into()));
+        let is_padding = circuit_builder.input(CurrentRow(IsPadding.into()));
+        let is_padding_next = circuit_builder.input(NextRow(IsPadding.into()));
```

Comment:
I thought that `ColumnId.into()` should be avoided in order to not sidestep something that might be important. Has that changed?